plugins/catalog-backend-module-gitlab/src/lib/client.ts

```diff
@@ -91,6 +97,163 @@ export class GitLabClient {
     return this.pagedRequest(`/groups`, options);
   }
 
+  async listSaasUsers(groupPath: string): Promise<PagedResponse<GitLabUser>> {
+    const items: GitLabUser[] = [];
+    let hasNextPage: boolean = false;
+    let endCursor: string | null = null;
+
+    if (!groupPath) {
+      throw new Error(
+        `Missing required 'group' value in gitlab provider config.`,
+      );
+    }
+
+    do {
+      const response: GitLabSaasUsersResponse = await fetch(
+        `${this.config.baseUrl}/api/graphql`,
+        {
+          method: 'POST',
+          headers: {
+            ...getGitLabRequestOptions(this.config).headers,
+            ['Content-Type']: 'application/json',
+          },
+          body: JSON.stringify({
+            variables: { group: groupPath, endCursor },
+            query: `query($group: ID!, $endCursor: String) {
+              group(fullPath: $group) {
+                groupMembers(first: 100, relations: [DESCENDANTS], after: $endCursor) {
+                  nodes {
+                    user {
+                      id 
+                      username 
+                      commitEmail
+                      name
+                      state
+                      webUrl
+                      avatarUrl  
+                    }
+                  }
+                  pageInfo {
+                    endCursor
+                    hasNextPage
+                  }
+                }
+              }
+            }`,
+          }),
+        },
+      ).then(r => r.json());
+      if (response.errors) {
+        throw new Error(`GraphQL errors: ${JSON.stringify(response.errors)}`);
+      }
+
+      if (!response.data.group?.groupMembers?.nodes) {
+        this.logger.warn(
+          `Couldn't get members under ${groupPath}. The provided token might not have sufficient permissions`,
+        );
+        continue;
+      }
+
+      const usersData = response.data.group.groupMembers.nodes;
+
+      for (let i = 0; i < usersData.length; i++) {
+        const userItem = usersData[i];
+
+        const formattedUserResponse = {
+          id: Number(userItem.user.id.replace(/^gid:\/\/gitlab\/User\//, '')),
+          username: userItem.user.username,
+          email: userItem.user.commitEmail,
+          name: userItem.user.name,
+          state: userItem.user.state,
+          web_url: userItem.user.webUrl,
+          avatar_url: userItem.user.avatarUrl,
+        };
+
+        items.push(formattedUserResponse);
+      }
+      ({ hasNextPage, endCursor } = response.data.group.groupMembers.pageInfo);
+    } while (hasNextPage);
+    return { items };
+  }
+
+  async listSaasGroups(groupPath: string): Promise<PagedResponse<GitLabGroup>> {
+    const items: GitLabGroup[] = [];
+    let hasNextPage: boolean = false;
+    let endCursor: string | null = null;
+
+    if (!groupPath) {
+      throw new Error(
+        `Missing required 'group' value in gitlab provider config.`,
```

Comment:
I wonder can we improve this error to include the provider name, for which the error is thrown, and also point out that this is when `host == gitlab.com`. See here - https://github.com/backstage/backstage/blob/master/plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts#L88-L94